### ISSUE TITLE
Split the rule that checks using `*()`, `*[]` and `**{}` in calls into different rules

### ISSUE THREAD

**Issue by GreyZmeem** [wemake-services/wemake-python-styleguide#767]
# Rule request

## Thesis

Divide the [WPS517](https://github.com/wemake-services/wemake-python-styleguide/blob/9105869ee4d38775ffbf4348ef29492794c1177f/wemake_python_styleguide/violations/refactoring.py#L704) rule into different rules, which individually check `*()` /  `*[]` and `**{}` expressions.

Related issue: https://github.com/wemake-services/wemake-python-styleguide/issues/715

## Reasoning

`**{}` can be used when the variable contains the name of the parameter, which should be passed to some method.  

For example:

```python
def user_by_username(username: str):
    return User.objects.get(**{User.USERNAME_FIELD: username})
```

To satisfy the linter, the code can be rewritten like this:

```python
def user_by_username(username: str):
    kwargs = {User.USERNAME_FIELD: username}
    return User.objects.get(**kwargs)
```

But I think it would be better if I could disable the linter check specifically for the `**{}` expression.

**Comment by sobolevn** [comment]
Yes, @GreyZmeem. I agree. We need to check `**{}` differently.

**Comment by orsinium** [comment]
As stayed in #589, we can recommend using `dict(...)` instead of `{...}` if all keys are correct literals not only for unpacking :thinking:

**Comment by orsinium** [comment]
Keys must be not just strings to use it as keyword arguments for dict(). It also must be srt.isidentifier()

**Comment by sobolevn** [comment]
Yes, good catch!

**Comment by sobolevn** [comment]
Ok, that's how we will handle it: for now all string keys will be reported as `WPS517`
Later we will also check that `print(**{'@': 2})` is a separate violation: https://github.com/wemake-services/wemake-python-styleguide/issues/782
